### DISCUSSION TITLE
Steps component returns numbers when clicked on a Step. Would be better to return the key of the Step.

### DISCUSSION THREAD

**Discussion by tanapaydin** [ant-design/ant-design#32471] [Ideas]
Steps component returns numbers when clicked on a Step, such as 0, 1, 2, 3... Would be better to return the given key of the Step instead.

We're trying to build a wizard, and the steps on that wizard are not predetermined. Depending on what a user does on the first step, e.g., the second step can be either componentA or componentB. And the number of total steps varies depending on the user actions on these steps. In short, the wizard is totally dynamic.

We're either showing a component on a step or removing it from dom if need be. Therefore, the index of a specific step varies. That is componentA can sometimes be the second step, and sometimes be the third step. Therefore, when we click on a step, instead of returning its index (2, 3) what we need is the key of that step. This way we can always make sure which specific step is clicked (when that step is visible).

Hope this request makes sense. This would make the Step component more useful.

**Comment by Angelii** [comment]
I also met with the same problem
